plugins/analytics/public/threatminer.py

```diff
@@ -50,50 +72,29 @@ def analyze(observable, results):
         links = set()
         params = {"q": observable.value, "rt": 1}
         json_result = ThreatMinerApi.fetch(observable, params, "sample.php")
-        if json_result:
-            json_string = json.dumps(
-                json_result,
-                sort_keys=True,
-                indent=4,
-                separators=(",", ": ")
-            )
-
-            result = dict(
-                [
-                    ("raw", json_string),
-                    ("source", "threatminer_query")
-                ]
-            )
-            results.update(raw=json_string)
-            _results = json_result.get("results")
 
-            if not _results:
-                return list(links)
-
-            for r in _results:
-                hashes = {
-                    "md5": r["md5"],
-                    "sha1": r["sha1"],
-                    "sha256": r["sha256"]
-                    }
+        _results, result = aux_checker(json_result)
+        if not _results:
```

Comment:
if you raise a GenericYetiError in `aux_checker`, this check (as well as all the others) is no longer necessary